File: 2_Number_Theory/not_accepted_and_tests/rationalarithmetic.java
import java.util.Scanner;
import java.math.BigInteger;
public class rational_arithmetic_java {

public static void main(String[] args) {
Scanner scan = new Scanner(System.in);

int n = scan.nextInt();

for (int i = 0; i < n; i++)
    {
    BigInteger x1 = new BigInteger(scan.next());
    BigInteger y1 = new BigInteger(scan.next());

    String operand = scan.next();

    BigInteger x2 = new BigInteger(scan.next());
    BigInteger y2 = new BigInteger(scan.next());

    BigInteger r1;
    BigInteger r2;

    if (operand.equals("+"))
        {
        r1 = (x1.multiply(y2)).add( x2.multiply(y1));
        r2 = y1.multiply(y2);
        }
    else if (operand.equals("-"))
        {
        r1 = (x1.multiply(y2)).subtract(x2.multiply(y1));
        r2 = y1.multiply(y2);
        }
    else if (operand.equals("*"))
        {
        r1 = x1.multiply(x2);
        r2 = y1.multiply(y2);
        }
    else
        {
        r1 = x1.multiply(y2);
        r2 = y1.multiply(x2);
        }

    BigInteger ggt = r1.gcd(r2);
    r1 = r1.divide(ggt);
    r2 = r2.divide(ggt);

    if (r2.compareTo(BigInteger.ZERO) < 0)
        {
        r2 = r2.negate();
        r1 = r1.negate();
        }

    System.out.println(r1 + " / " + r2);
    }

scan.close();
    }
}

File: 2_Number_Theory/not_accepted_and_tests/rationalarithmetic.py
from math import gcd


def kuerzen(p1, p2):
    ggt = gcd(p1, p2)
    r1 = p1 / ggt
    r2 = p2 / ggt
    if r2 < 0:
        r1 *= -1
        r2 *= -1
    return r1, r2


def ausgabe(r1, r2):
    print(f"{int(r1)} / {int(r2)}")


def add_sub(x1, y1, x2, y2, op):
    b1 = x1 * y2
    b2 = x2 * y1
    p2 = y1 * y2
    if op == "+":
        p1 = b1 + b2
    else:
        p1 = b1 - b2
    r1, r2 = kuerzen(p1, p2)
    ausgabe(r1, r2)


def mult(x1, y1, x2, y2):
    p1 = x1 * x2
    p2 = y1 * y2
    r1, r2 = kuerzen(p1, p2)

    ausgabe(r1, r2)


for _ in range(int(input())):
    x1, y1, op, x2, y2 = input().split()
    if op in ["+", "-"]:
        add_sub(int(x1), int(y1), int(x2), int(y2), op)
    elif op == "*":
        mult(int(x1), int(y1), int(x2), int(y2))
    else:
        mult(int(x1), int(y1), int(y2), int(x2))
